### ISSUE TITLE
Path parameter of type date-time should be url encoded

### ISSUE THREAD

**Issue by cws-alb** [dart-ogurets/dart-openapi-maven#34]
If a path paramter contains a path parameter definition of date-time, the string generated in the toString() should be url encoded to ensure the space and any other characters should be properly handled. Currently, it seems dio is not happy with a path contiaining illegal url characters such as space.

```
  /endpoint/with/paths/{id}/{interesting-time}:
...
  parameters:
        - $ref: '#/components/parameters/path-id'
        - name: interesting-time
          in: path
          required: true
          schema:
            type: string
            format: date-time
            maxLength: 26
            example: '2020-06-30T22:00:00+01:00'
```

**Comment by rvowles** [comment]
this is a very quirky API, but fair enough!

**Comment by rvowles** [comment]
This is getting awkward to support in both null-safe and non-null-safe environments. Further, Date's are not encoded correctly, they are encoded as datetimes when being passed as query parameters. Are you null-safe or not?

**Comment by rvowles** [comment]
same branch as the multipart file, I also have a working release for this.

**Comment by rvowles** [comment]
resolved in 5.5

**Comment by gbloggs** [comment]
quality!
